rust-toolchain.toml

```diff
@@ -0,0 +1,4 @@
+[toolchain]
```

Comment:
nice!

Reply:
For now our builds will continue to use the legacy `rust-toolchain` by default.  There are a couple of incremental paths forward:
- if we update `rustup` on the CI we can rename `rust-toolchain.toml` to `rust-toolchain`
- alternatively (and without the need to update `rustup`) if we remove from the CI the expectation for a `rust-toolchain` file we can just have `rust-toolchain.toml`, which I think is the current best practice